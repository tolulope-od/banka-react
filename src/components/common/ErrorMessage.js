import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({ errors }) => (
  <div>
    <img
      src="https://res.cloudinary.com/tolulope-od/image/upload/v1563463864/action-error_ohkpfa.png"
      alt="error-image"
      width="80"
    />
    <h3>{errors.error || 'Something Unusual happened'}</h3>
    <input
      type="submit"
      className="accnt-create-btn"
      name="addAccount"
      value="OK"
      id="submit-btn"
    />
  </div>
);

ErrorMessage.propTypes = {
  errors: PropTypes.object
};

export default ErrorMessage;
