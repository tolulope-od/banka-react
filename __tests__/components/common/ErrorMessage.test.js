import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorMessage from '../../../src/components/common/ErrorMessage';

const shallowSetup = () => {
  const props = {
    errors: { error: 'Something happend, request not processed' }
  };

  const enzymeWrapper = shallow(<ErrorMessage {...props} />);

  return { props, enzymeWrapper };
};

describe('<ErrorMessage /> Component', () => {
  const { props } = shallowSetup();

  it('Renders the Error Message component', () => {
    const wrapper = shallow(<ErrorMessage {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should return an active id on the dashboard icon it is active', () => {
    const wrapper = shallow(<ErrorMessage {...props} />);
    expect(wrapper.find('input').length).toEqual(1);
    expect(wrapper.find('h3').length).toEqual(1);
    expect(wrapper.find('img').length).toEqual(1);
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('h3').text()).toEqual('Something happend, request not processed');
  });

  it('renders the default error message when no error message is supplied', () => {
    const wrapper = shallow(<ErrorMessage {...props} />);
    wrapper.setProps({
      errors: {}
    });
    expect(wrapper.find('h3').text()).toEqual('Something Unusual happened');
  });
});
