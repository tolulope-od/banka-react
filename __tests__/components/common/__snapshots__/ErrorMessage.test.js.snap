// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> Component Renders the Error Message component 1`] = `
<div>
  <img
    alt="error-image"
    src="https://res.cloudinary.com/tolulope-od/image/upload/v1563463864/action-error_ohkpfa.png"
    width="80"
  />
  <h3>
    Something happend, request not processed
  </h3>
  <input
    className="accnt-create-btn"
    id="submit-btn"
    name="addAccount"
    type="submit"
    value="OK"
  />
</div>
`;
